modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js

```diff
@@ -48,8 +48,9 @@ const RichText = ({
 
 			editor.config.contentsLanguage = editingLanguageId;
 
-			editor.setData(editor.getData());
+			editor.setData(contents);
```

Comment:
This change, along with no addition of `contents` in dependencies, means that this code will execute only once. Last time this update was removed, there was a regression, fixed in https://github.com/liferay-forms/liferay-portal/pull/945. The related tickets is https://issues.liferay.com/browse/LPS-139520

Reply:
> This change, along with no addition of contents in dependencies, means that this code will execute only once

I agree that it will only execute once, but I think using

```js
editor.setData(editor.getData());
```

is also wrong: `editor.getData()` [returns the data in the editor](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#method-getData), so when reading this line it looks like we're setting the data to what it previously was. On top of that, when debugging this part of the code, you'll notice that `editor.state` returns `unloaded` which is why the data doesn't get set properly. Maybe it's time to review the entire component with the team because I have the feeling we have being going multiple times through this bug.

Reply:
Tested the text direction with this PR's changes, it's working, so no regression here! 🥳 

https://user-images.githubusercontent.com/6642927/141309009-ffd3be48-2d79-4c24-9c8c-cad4270fbec7.mov


The effect will run whenever the `editingLanguageId` prop changes, which happens when you select a new language. Before this fix, the effect was calling `editor.setData` with whatever the editor contained, which was the previous language. So it was setting the new language's content to the previous one (you can see the result in the first video in the PR description). We don't need `contents` as an effect dependency because it changes at the same time as `editingLanguageId`. 

If I add the contents to the dependency array, this is the result:

https://user-images.githubusercontent.com/6642927/141309931-b4fe8b2f-a1af-4d38-bf0a-1ed2451ad741.mov

It appears that calling `editor.setData` moves the caret to the start of the editor.

Reply:
After looking at [these bugs](https://issues.liferay.com/browse/PTR-2589) (thanks @markocikos!) I saw that changes to the predefined value of the `RichText` weren't displayed. Adding the `predefinedValue` as a dependency to this effect fixes that, without causing the bug in the 2nd video. But it looks like a ñapa 😄